src/window-event-handler.coffee

```diff
@@ -24,13 +23,12 @@ class WindowEventHandler
             if pathToOpen? and needsProjectPaths
               if fs.existsSync(pathToOpen)
                 atom.project.addPath(pathToOpen)
-              else if fs.existsSync(path.dirname(pathToOpen))
-                atom.project.addPath(path.dirname(pathToOpen))
               else
-                atom.project.addPath(pathToOpen)
+                dirToOpen = path.dirname(pathToOpen)
+                if fs.existsSync(dirToOpen)
+                  atom.project.addPath(dirToOpen)
 
-            {protocol} = url.parse(pathToOpen)
-            unless fs.isDirectorySync(pathToOpen) or protocol?
+            unless fs.isDirectorySync(pathToOpen)
```

Comment:
1- will check for the undefined and fix it
2- how about: if the uri starts with `atom://`, then it needs to be opened.
or how do I differentiate between the reload/restart case and the normal open cases?